### PR TITLE
SMILE-1290: Corresponding author set on contributor

### PR DESCRIPTION
**Pull request by nraanes**
Leser inn CorrespondenceTp indexedName fra Scopus-xml og setter correspondingAuthor til true på Contributor dersom match med indexedName på AuthorTp

### CODE REVIEW

**Review by anetteOlli — APPROVED** [review]

File: create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusConverter.java [hunk 1]
```diff
@@ -387,25 +406,38 @@ private boolean isTitleOriginal(TitletextTp titletextTp) {
         return extractHead().getCitationTitle().getTitletext();
     }
 
-    private List<Contributor> generateContributorsFromAuthorGroup(AuthorGroupTp authorGroupTp) {
+    private List<Contributor> generateContributorsFromAuthorGroup(AuthorGroupTp authorGroupTp,
+                                                                  PersonalnameType correspondencePerson) {
         return authorGroupTp.getAuthorOrCollaboration()
             .stream()
-            .map(this::generateContributorFromAuthorOrCollaboration)
+            .map(o -> generateContributorFromAuthorOrCollaboration(o, correspondencePerson))
```
Comment (brinxmat) [inline create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusConverter.java:413]: What is an 'o'?

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java [hunk 2]
```diff
@@ -627,16 +622,41 @@ void shouldNotThrowExceptionWhenDoiInScopusIsNull() throws IOException {
...
-        assertThat(actualPublicationInstance.getIssue(), is(EXPECTED_ISSUE_IN_0000469852));
-        assertThat(actualPublicationInstance.getPages().getBegin(), is(EXPECTED_FIRST_PAGE_IN_0000469852));
-        assertThat(actualPublicationInstance.getPages().getEnd(), is(EXPECTED_LAST_PAGE_IN_0000469852));
+        assertThat(actualPublicationInstance.getVolume(), is(expectedVolume));
+        assertThat(actualPublicationInstance.getIssue(), is(expectedIssue));
+        assertThat(actualPublicationInstance.getPages().getEnd(), is(expectedPages));
+    }
+
+    @Test
+    void shouldExtractCorrespondenceAndSetMatchingContributorToCorrespondingAuthor() throws IOException {
```
Comment (brinxmat) [inline create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:641]: Is this "shouldExtractCorrespondingAuthors"?

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusConverter.java (hunk 1 above)
Comment (nraanes) [reply create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusConverter.java:413]: It's the circle around your head 👼 (120 line limit makes me invent names like 'o').

File: create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusConverter.java [hunk 3]
```diff
@@ -414,34 +414,67 @@ private boolean isTitleOriginal(TitletextTp titletextTp) {
...
+        return List.of(organization);
+    }
+
+    private Optional<Map<String, String>> getOrganizationLabels(AuthorGroupTp authorGroup) {
+        var organizationNameOptional = getOrganizationNameFromAuthorGroup(authorGroup);
+        return organizationNameOptional.map(organizationName -> Map.of(getLanguageIso6391Code(),
+                                                                       organizationName));
+    }
+
+    private String getLanguageIso6391Code() {
```
Comment (brinxmat) [inline create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusConverter.java:457]: Here, you're not handling more than one language?

Comment (brinxmat) [reply create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusConverter.java:457]: You can also use the Language constants from nva-language-java

**Review by brinxmat — CHANGES_REQUESTED** [review]

**Review by gregorgabriel — APPROVED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**